### DISCUSSION TITLE
How to open PDF with SEB 2.3.2 in macOS

### DISCUSSION THREAD

**Discussion by aurelienb1** [SafeExamBrowser/seb-mac#125] [Q&A] [answered]
Hi,

The opening of PDF works perfectly for Windows because we can authorize the Acrobat Reader application.

How to do it under mac, it displays me missing plugin or a blank page depending on the configuration:

- Download and open PDF files instead of displaying them inline = blank page
- Allow using Acrobat Reader PDF plugin (insecure) = missing plugin

How to install the plugin, I installed FireFox and indicated that I wanted the pdf to open in FireFox but it does not change anything.

I am not familiar with the macOS environment.

Regards

**Comment by danschlet** [comment, marked as answer]
On current macOS versions, it's not possible to use the Acrobat reader plugin anymore, as Apple doesn't support it (also not in Safari). But in one case previously the installed plugin still was interfering with SEB, but as far as I remember, there was a bug in a previous version I fixed. 

To use the Acrobat Reader application is not a good idea in secure exams, as students can access PDF files on their local and cloud drives in the app.

If you leave those two settings in the default disabled position, PDFs are displayed by the browser engine. If it doesn't work on your webpage, then they are most likely embedded in a non-standard way. Test it with some standard HTML page, for example the "SEB 2.x How To" PDF on the manual pages on safeexambrowser.org should work perfectly.

Also make sure you don't have a URL filter set which would block the PDF display. Maybe the [SEB logs](https://safeexambrowser.org/macosx/mac_usermanual_en.html#SecurityPane) show some error when the URL of the PDF should be loaded.

If you don't figure it out, I could have a look if you can give me access to the page where it doesn't work.

If it's some web API incompatibility, it might work with SEB 3.0, see the beta [release](https://github.com/SafeExamBrowser/seb-mac/releases).

**Comment by aurelienb1** [comment]
Hi Daniel,

Finally by unchecking the options I am able to open the pdf natively in Safe Exam Browser in MacOS.

The problem is that our LMS only allows to download the pdf, not to display them in the browser.

We can close the discussion as it is functional and the problem is more with our LMS.

Regards

**Reply by danschlet** [reply]
There might be improvements with SEB 3.0, I still need to test if displaying PDFs works better with the modern WebView (although existing LMS integrations might need to be updated for SEB being able to use the modern WebView). Moodle is such a case. Are you using an LMS called "note LMS", I never heard of that?

**Comment by aurelienb1** [comment]
LMS = learning management system, in our case Moodle.

Regards

**Reply by danschlet** [reply]
I know the abbreviation but I misunderstood your sentence. Thanks for clarifying!

**Reply by aurelienb1** [reply]
Sorry, my mistake, my brain thinks in French while I type in English :)
